src/Indigo/Contracts/Holdings/Impl.hs

```diff
@@ -0,0 +1,195 @@
+          (storage !. #fields) ~. (#mbNewAdmin, some $ newAdmin)
+    , #cAcceptAdminRights //-> \_unit -> do
+        doc $ DDescription "Accept admin rights by the new admin."
+        ifJust (storage !. #fields !. #mbNewAdmin)
+          (\newAdmin -> do
+              if newAdmin /=. Sender
+                then () <$ failCustom_ #senderIsNotNewAdmin
+                else do
+                setField_ storage #fields $ (storage !. #fields) ~. (#admin, newAdmin)
+                setField_ storage #fields $ (storage !. #fields) ~. (#mbNewAdmin, none)
+          )
+          (() <$ failCustom_ #notInTransferAdminRightsMode)
+    , #cIsAdmin //-> \v -> do
+        doc $ DDescription
+          "Check whether address is admin. Returns `True` if the address is an admin."
+        flip view_ v $ \addr -> return $ (storage !. #fields !. #admin) ==. addr
+    , #cTransfer //-> \tParams -> do
+        ensureSenderIsAdmin
+        -- pause check is already done in Lorentz code
+        ensureIsTransferable
+        let from = tParams !. #from
+            to = tParams !. #to
+        callAssertTransfer $ pair (Name #from from) (Name #to to)
+        transfer tParams
+    , #cApprove //-> \aParams -> do
+        -- pause check is already done in Lorentz code
+        let spender = aParams !. #spender
+            receivers = spender .: sender .: nil
+        callAssertReceivers receivers
+        approve aParams
+    , #cGetAllowance //-> getAllowance
+    , #cGetBalance //-> getBalance
+    , #cGetTotalSupply //-> getTotalSupply
+    , #cMint //-> \mParams -> do
+        ensureNotPaused
+        -- admin check is already done in Lorentz code
+        let to = mParams !. #to
+        callAssertReceiver to
+        mint mParams
+    , #cBurn //-> \bParams -> do
+        ensureNotPaused
+        -- admin check is already done in Lorentz code
+        burn bParams
+    , #cBurnAll //-> \_ -> do
+        doc $ DDescription "Destroy all tokens and allowances."
+        ensureSenderIsAdmin
+        ensureNotPaused
+        setField_ storage #ledger emptyBigMap
+    , #cSetPause //-> \paused -> do
+        -- admin check is already done in Lorentz code
+        setPause paused
+    , #cSetTransferable //-> \transferable -> do
+        doc $ DDescription "Change transferable flag."
+        ensureSenderIsAdmin
+        setField_ storage #fields $
+          (storage !. #fields) ~. (#transferable, transferable)
+    , #cGetTokenMeta //-> \v -> do
+        doc $ DDescription "Get token meta data: name, symbol and id."
+        flip view_ v $ \_ -> return (storage !. #fields !. #tokenMeta)
+    )
+
+callAssertTransfer
+  :: (HasStorage Storage, HasSideEffects, a :~> ("from" :! Address, "to" :! Address))
+  => a -> IndigoProcedure s
+callAssertTransfer a = do
+  whenJust (storage !. #fields !. #mbSafelistAddress)
+    (\addr -> do
+        ifJust (contractCallingUnsafe @("from" :! Address, "to" :! Address)
+          (eprName $ Call @"assertTransfer") addr)
```

Comment:
Confusing indentation, this looks like the first branch of the `ifJust` (same below)